src/devices/secplus_v2.c

```diff
@@ -0,0 +1,555 @@
+/** @file
+    Security+ 2.0 rolling code
+
+    Copyright (C) 2020 Peter Shipley <peter.shipley@gmail.com>
+    Based on code by Clayton Smith https://github.com/argilo/secplus
```

Comment:
License-wise, this is not fully compatible, as the code this is based on is GPL-3 instead of GPL-2 :(.

I didn't look into how 'copied' this looks, but keep that in mind. If it's all your own and loosely based upon the original, it shouldn't matter much (but IANAL).